nvflare/private/fed/utils/app_deployer.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+
+from nvflare.apis.job_def import JobMetaKey
+from nvflare.apis.workspace import Workspace
+from nvflare.fuel.hci.zip_utils import unzip_all_from_bytes
+from nvflare.private.privacy_manager import PrivacyService
+
+from .app_authz import AppAuthzService
+
+
+class AppDeployer(object):
+    def __init__(self, workspace: Workspace, job_id: str, job_meta: dict, app_name: str, app_data):
+        self.app_name = app_name
+        self.workspace = workspace
+        self.app_data = app_data
+        self.job_id = job_id
+        self.job_meta = job_meta
+
+    def deploy(self) -> str:
+        """
+        Try to deploy the app.
+
+        Returns: error message if any
+
+        """
+        job_scope = self.job_meta.get(JobMetaKey.SCOPE, "")
+
+        # check whether this scope is allowed
+        if not PrivacyService.is_scope_allowed(job_scope):
+            return f"job scope '{job_scope}' is not allowed"
+
+        try:
+            run_dir = self.workspace.get_run_dir(self.job_id)
+            app_path = self.workspace.get_app_dir(self.job_id)
+            app_file = os.path.join(run_dir, "fl_app.txt")
+            job_meta_file = self.workspace.get_job_meta_path(self.job_id)
+
+            if os.path.exists(run_dir):
+                shutil.rmtree(run_dir)
+
+            if not os.path.exists(app_path):
+                os.makedirs(app_path)
+
+            unzip_all_from_bytes(self.app_data, app_path)
+
+            with open(app_file, "wt") as f:
+                f.write(f"{self.app_name}")
+
+            with open(job_meta_file, "w") as f:
+                json.dump(self.job_meta, f, indent=4)
+
+            assert isinstance(self.job_meta, dict)
```

Comment:
if we need this check we should check in `__init__` and use if instead of assert